File: Concurrent/phoenix_concurrent/lib/phoenix_concurrent_web/controllers/concurrent_controller.ex
defmodule PhoenixConcurrentWeb.ConcurrentController do
  use PhoenixConcurrentWeb, :controller

  def readFile(conn, _params) do
    file_to_read = "textFile.txt"

    task = Task.async(fn -> read_file(file_to_read) end)
    IO.inspect(task.pid, label: "New Task PID")

    result = Task.await(task, 5000)

    case result do
      {:ok, content} ->
        json(conn, %{success: true, message: content})

      {:error, reason} ->
        json(conn, %{success: false, message: "Error reading file: #{reason}"})
    end
  end

  defp read_file(file) do
    try do
      file_path = Path.join([:code.priv_dir(:phoenix_concurrent), "static", file])
      file_content = File.read!(file_path)

      :timer.sleep(2000)

      {:ok, file_content}
    rescue
      e ->
        {:error, e.message}
    end
  end
end

File: Concurrent/Laravel/app/Http/Controllers/ConcurrentController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Log\Logger;
use Illuminate\Support\Facades\Concurrency;

class ConcurrentController extends Controller
{
    public function readFile(Request $request)
    {
        try {
            $filePath = storage_path('app/textFile.txt');
            if (!file_exists($filePath)) {
                return response()->json(['error' => 'File does not exist'], 404);
            }
            try {
                $fileContent = Concurrency::run(function () use ($filePath) {
                    $_filecontent =  file_get_contents($filePath);
                    sleep(2);
                    return $_filecontent;
                });
                return response()->json(['success' => true, 'message' => $fileContent]);
            } catch (\Exception $e) {
                return response()->json(['success' => false, 'message' => 'Error reading file: ' . $e->getMessage()], 500);
            }
        } catch (\Exception $e) {
            return response()->json(['error' => $e->getMessage()], 500);
        }
    }
}
